### PR TITLE
Xchacha20

### PR DESCRIPTION
**Pull request by rklaehn**
Use XChaCha20 instead of XSalsa20

this is in case we don't do the more complex thing of replacing nonces with seeks.

### CODE REVIEW

**Comment by rklaehn** [comment]
As for motivation: chacha20 has a bit better cryptographic properties than salsa20. Due to this, it has been more widely adopted, see https://en.wikipedia.org/wiki/Salsa20#ChaCha_variant , and has some more high performance and better audited libraries available.

Nothing really wrong with salsa20 though. It is not borken or anything...

**Review by wngr — APPROVED** [review]

File: banyan/src/zstd_dag_cbor_seq.rs [hunk 1]
```diff
@@ -246,19 +248,25 @@ impl ZstdDagCborSeq {
...
-        key: &salsa20::Key,
+        nonce: &chacha20::XNonce,
+        key: &chacha20::Key,
     ) -> anyhow::Result<Vec<u8>> {
         let Self { mut data, links } = self;
         // encrypt in place with the key and nonce
-        XSalsa20::new(key, nonce).apply_keystream(&mut data);
+        let mut chacha20 = XChaCha20::new(key, nonce);
+        chacha20.seek(0u64);
+        chacha20.apply_keystream(&mut data);
```
Comment (dvc94ch) [inline banyan/src/zstd_dag_cbor_seq.rs:269]: not sure we need the seek

**Review by dvc94ch — APPROVED** [review]
LGTM

File: banyan/src/zstd_dag_cbor_seq.rs (hunk 1 above)
Comment (rklaehn) [reply banyan/src/zstd_dag_cbor_seq.rs:269]: we don't. Was a leftover from the more ambitious stuff...

**Pull request merged**